File: challenging/18_neuron/neuron_test.py
import unittest
from neuron import Neuron
import numpy as np

class TestTournamentWinner(unittest.TestCase):
    def test_tournament_winner(self):
        examples = [
        		{"features": [0, 0], "label": 0.0},
        		{"features": [0, 1], "label": 1.0},
        		{"features": [1, 0], "label": 1.0},
        		{"features": [1, 1], "label": 0.0},
        	]
        neuron = Neuron(examples)
        actual = np.round(neuron.predict([0.79, 0.89, 0.777]))
        expected = 1.0
        self.assertEqual(actual, expected)

File: challenging/18_neuron/neuron_test.go
package main

import (
	"github.com/stretchr/testify/require"
	"math"
	"testing"
)

func TestTournamentWinner(t *testing.T) {
	examples := []Example{
		Example{
			Features: []float64{0.0, 0.0},
			Label:    0.0,
		},
		Example{
			Features: []float64{0.0, 1.0},
			Label:    1.0,
		},
		Example{
			Features: []float64{1.0, 0.0},
			Label:    1.0,
		},
		Example{
			Features: []float64{1.0, 1.0},
			Label:    0.0,
		},
	}
	neuron := Neuron{
		weights:  []float64{0.49671415, 0.64768854, 0.52302986, 0.64589411},
		Examples: examples,
	}
	neuron.Train(0.01, 10, 200)
	actual := math.Round(neuron.Predict([]float64{0.79, 0.89, 0.777}))
	expected := 1.0
	require.Equal(t, expected, actual)
}
